### PR TITLE
Fix handling of content items

### PR DESCRIPTION
**Pull request by tmeschter**
Fixes https://github.com/dotnet/project-system/issues/3042.

Consider the following content files from a sample project.assets.json file:

``` JSON
  "targets": {
    ".NETFramework,Version=v4.7.2": {
      "DryIocZero/4.0.0-preview-04": {
        "type": "package",
        "contentFiles": {
          "contentFiles/cs/any/Container.Generated.tt": {
            "buildAction": "None",
            "codeLanguage": "cs",
            "copyToOutput": false
          },
          "contentFiles/cs/any/Container.cs": {
            "buildAction": "Compile",
            "codeLanguage": "cs",
            "copyToOutput": false
          },
          "contentFiles/cs/any/ImTools.cs": {
            "buildAction": "Compile",
            "codeLanguage": "cs",
            "copyToOutput": false
          },
          "contentFiles/cs/any/Registrations.ttinclude": {
            "buildAction": "None",
            "codeLanguage": "cs",
            "copyToOutput": false
          },
          "contentFiles/any/any/MyAwesomeFile.txt": {
            "buildAction": "None",
            "codeLanguage": "any",
            "copyToOutput": false
          },
          "contentFiles/cs/net472/Alpha.cs": {
            "buildAction": "Compile",
            "codeLanguage": "cs",
            "copyToOutput": false
          },
          "contentFiles/any/net472/AnyNet472.txt": {
            "buildAction": "None",
            "codeLanguage": "any",
            "copyToOutput": false
          }
        },
```

Currently, in a C# project we'll only create MSBuild items for three of these:

* contentFiles/cs/any/Container.cs
* contentFiles/cs/any/ImTools.cs
* contentFiles/cs/net472/Alpha.cs

I believe the correct behavior is to create MSBuild items for all seven.

There are two overlapping issues:

1. We simply ignore any item that specifies `"buildAction": "None"` on the assumption that these items do not participate in the build (which is a bad assumption, and out-of-sync with what happens in SDK-style projects).
2. If any item in `contentFiles` has a `codeLanguage` matching the project language (e.g. C#, VB, F#, etc.) then we ignore those with `"codeLanguage": "any"`. I believe the correct behavior is to always pick up all the files marked with `"codeLanguage": "any"`, _plus_ all the items with a `codeLanguage` that matches the project language.

### CODE REVIEW

File: src/Microsoft.NuGet.Build.Tasks/Microsoft.NuGet.Build.Tasks.csproj [hunk 1]
```diff
@@ -9,7 +9,7 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>Microsoft.NuGet.Build.Tasks</RootNamespace>
     <AssemblyName>Microsoft.NuGet.Build.Tasks</AssemblyName>
-    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.7.2</TargetFrameworkVersion>
```
Comment (jjmew) [inline src/Microsoft.NuGet.Build.Tasks/Microsoft.NuGet.Build.Tasks.csproj]: ><TargetFrameworkVersion>v4.7.2</TargetFrameworkVersion> [](start = 4, length = 55)

This is a silly question but are we sure that the build machines have 4.7.2 installed?

Comment (tmeschter) [reply src/Microsoft.NuGet.Build.Tasks/Microsoft.NuGet.Build.Tasks.csproj]: Yep.

**Comment by davkean** [comment]
tag @nkolev92

File: src/Microsoft.NuGet.Build.Tasks/Microsoft.NuGet.Build.Tasks.csproj [hunk 2]
```diff
@@ -70,6 +69,14 @@
       <LastGenOutput>Strings.Designer.cs</LastGenOutput>
     </EmbeddedResource>
   </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Build.Utilities.Core">
+      <Version>14.3.0</Version>
+    </PackageReference>
+    <PackageReference Include="Newtonsoft.Json">
```
Comment (nkolev92) [inline src/Microsoft.NuGet.Build.Tasks/Microsoft.NuGet.Build.Tasks.csproj]: Why not use 9.0.1 like the rest of the components?

Comment (nkolev92) [reply src/Microsoft.NuGet.Build.Tasks/Microsoft.NuGet.Build.Tasks.csproj]: I get that this is just moving it to PackageReference, but if you're already moving this project to net472 🤷‍♂

Comment (tmeschter) [reply src/Microsoft.NuGet.Build.Tasks/Microsoft.NuGet.Build.Tasks.csproj]: The version here doesn't actually matter. For historical reasons, these project files are only used to build on a local development machine. The project files that build the actual binaries we ship are in the VS repo. We're going to fix that just as soon as we can...

**Comment by dmonroym** [comment]
We've moved things around this repo. Please reopen this PR against `master` if you want us to take a look at this again.

**Pull request closed without merging**